Add pipeline manager

MayoG opened the pull request.

MayoG commented:
The main thing to focus on here is the PipelineManager, most of the other files are routines that took from the components in contrib that we shouldn't use anymore. We still need to move more of the routines from these components but for now we have enough routines to have a full workflow of the pipe.

One more thing to note, I created a special folders for the routines and the unique components and the PiplineManager know to get them only from their folders, It is also impoetent that the names of these routines and components will be like that: 
Routine file name - my_routine.
Routine class name - MyRoutine
same for the unique components.

ItamarWilf commented:
We should add:
- [ ] Documentation of the config format for new routines / components.
- [ ] Tests including running multiple pipelines with different components and routines.

MayoG commented:
To make it easier to review this I will divide the code to seperate parts:

- pipeline_manager.py = Created the new manager and methods to control the entire pipe

- component.py = Deleted the zerorpc server and added new methods to control the queues and routines

- routine.py = Added routine type data member, method to get the parameters of each routine and changed the way the runner creating the threads for the routine.

- main.py = The file we will execute that will choose whether we want an API or CLI

- pipert/contrib/(components|routines).py = Moved the routines from the previous components and added them new methods that routine must implement such as get_constructor_parameters.

- I will add soon new tests for the pipeline manager

Sorry for this long pull request, didn't expect it to be that big when i started making it.
